### ISSUE TITLE
[Test] check if cephcsi is having any issue to support multiple accessModes.

### ISSUE THREAD

**Issue by Madhu-1** [ceph/ceph-csi#4853]
Kubernetes and the CSI spec support creating a volume with multiple accessModes, check if cephcsi is already supporting it or not and see if there are any issues with the supports, if it's supported and don't have any issues we need to add E2E testing for the same.

**Comment by iPraveenParihar** [comment]
Tried with below possible combinations - 

- RWO, RWX - ✅ 
- (ROX, RWX) & (ROX, RWO) & (ROX, RWX, RWO) - ❌ failed to provision volume with StorageClass "rook-ceph-block": rpc error: code = InvalidArgument desc = readOnly accessMode is supported only with content source

we have this check for RWO here - 
https://github.com/ceph/ceph-csi/blob/d8a040719e0ad491ad144fe361b67e6bd2109710/internal/util/validate.go#L88-L99

probably, we need to remove this check or tweak the condition to allow above combination?

**Comment by Madhu-1** [comment]
ROX is only supported for clone from PVC or clone from snapshot not for a fresh PVC because its empty PVC and of no use for read only mode.

**Comment by iPraveenParihar** [comment]
> ROX is only supported for clone from PVC or clone from snapshot not for a fresh PVC because its empty PVC and of no use for read only mode.

I agree, but now that we aim to support multiple access modes, this condition is no longer relevant. What about a PVC with both ROX and RWO access modes, where one pod writes data while others access it as read-only? 🤔

**Comment by Madhu-1** [comment]
> > ROX is only supported for clone from PVC or clone from snapshot not for a fresh PVC because its empty PVC and of no use for read only mode.
> 
> I agree, but now that we aim to support multiple access modes, this condition is no longer relevant. What about a PVC with both ROX and RWO access modes, where one pod writes data while others access it as read-only? 🤔

The kubernetes access needs to be translated into CSI specification, AFAIK we don't have such a cap in csi spec, if we have one we need to support it as well.

ROX translates to `VolumeCapability_AccessMode_MULTI_NODE_READER_ONLY` where its multi node reader only not one writer and many readers. lets dig more and see if we are missing anything and we can add such cases as we need to comply with CSI spec.

**Comment by Madhu-1** [comment]
https://github.com/kubernetes-csi/csi-lib-utils/blob/master/accessmodes/access_modes.go might help you to translate kube access to CSI spec
